datalad_next/iter_collections/annexworktree.py

```diff
@@ -0,0 +1,208 @@
+"""Report on the content of a Git-annex repository worktree
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from pathlib import (
+    Path,
+    PurePath,
+    PurePosixPath,
+)
+from typing import (
+    Any,
+    Generator,
+)
+
+from more_itertools import intersperse
+
+from .gitworktree import (
+    GitWorktreeItem,
+    GitWorktreeFileSystemItem,
+    FileSystemItemType,
+    iter_gitworktree
+)
+from datalad_next.itertools import (
+    itemize,
+    load_json,
+    route_in,
+    route_out,
+    StoreOnly,
+)
+from datalad_next.runners import iter_subproc
+
+
+lgr = logging.getLogger('datalad.ext.next.iter_collections.annexworktree')
+
+
+@dataclass
+class AnnexWorktreeItem(GitWorktreeItem):
+    annexkey: str | None = None
+    annexsize: int | None = None
+    annexobjpath: PurePath | None = None
+
+
+@dataclass
+class AnnexWorktreeFileSystemItem(GitWorktreeFileSystemItem):
+    annexkey: str | None = None
+    annexsize: int | None = None
+    annexobjpath: PurePath | None = None
+
+
+def content_path(item: AnnexWorktreeItem | AnnexWorktreeFileSystemItem,
+                 base_path: Path,
+                 link_target: bool,
+                 ) -> PurePath | None:
+
+    if item.annexobjpath:
+        return \
+            item.annexobjpath \
+            if (base_path / item.annexobjpath).exists() \
+            else None
```

Comment:
I believe this does not work. There is no guarantee that `base_path` is the root of the repository.

Reply:
I think it will work, because we use `-C base_path` when calling `git annex examinekey` (if `base_path` is not the root of the repository `git -C base_path annex examinekey` will yield an object-path so that `base_path / item.annexobjpath` will point to the annexed content).

But I might not be aware of all possible pitfalls here.

Reply:
You are corrrect